scripts/lighthouseRunner.js

```diff
@@ -0,0 +1,88 @@
+const lighthouse = require('lighthouse');
+const chromeLauncher = require('chrome-launcher');
+const chalk = require('chalk');
+
+const log = console.log;
+const config = require(`./../lighthouse`);
+
+function launchChromeAndRunLighthouse(url, opts, config = null) {
+  return chromeLauncher
+    .launch({ chromeFlags: opts.chromeFlags })
+    .then(chrome => {
+      opts.port = chrome.port;
+      return lighthouse(url, opts, config).then(results =>
+        chrome.kill().then(() => results.lhr),
+      );
+    });
+}
+
+function getScoresPerCategory(reportCategories, audits, url) {
+  const scores = Object.keys(reportCategories).map(category => ({
+    id: reportCategories[category].id,
+    score: reportCategories[category].score,
+  }));
+
+  return { scores, url, audits };
+}
+
+function validateScores(resultsArray) {
+  return resultsArray.map(result => ({
+    url: result.url,
+    scores: result.scores.map(actual => {
```

Comment:
Could you extract this to a function? Nested maps convolutes `validateScores` and increases the cognitive overheard.